applications: serial_lte_modem: add LwM2M Carrier Settings support (#13165)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst

------------

The test command is not supported.

LwM2M Carrier library configuration #XCARRIERCFG
================================================

> The ``#XCARRIERCFG`` command allows you to configure the LwM2M carrier library. The settings shall be stored and apply persistently.

Reviewer: ```suggestion
The ``#XCARRIERCFG`` command allows you to configure the LwM2M carrier library.
The settings are stored and applied persistently.
```

.. note::
   To use ``#XCARRIERCFG``, the :kconfig:option:`CONFIG_LWM2M_CARRIER_SETTINGS` Kconfig option must be enabled.

>    For more details on library configuration, refer to :ref:`lwm2m_configuration`.

Reviewer: ```suggestion
   For more details on LwM2M carrier library configuration, see the :ref:`lwm2m_configuration` section of the library's documentation.
```
Just a little more detail since the link will be titled just "Configuration".

>    For more details on library configuration, refer to :ref:`lwm2m_app_int:lwm2m_configuration`.

Reviewer 2: :ref:`lwm2m_configuration` should be enough to find the right section

* ``AT#XCARRIERCFG="coap_con_interval"[,<interval>]``
* ``AT#XCARRIERCFG="config_enable"[,<0|1>]``
* ``AT#XCARRIERCFG="device_enable"[,<0|1>]``
* ``AT#XCARRIERCFG="device_type"[,<device_type>]``
* ``AT#XCARRIERCFG="hardware_version"[,<version>]``
* ``AT#XCARRIERCFG="manufacturer"[,<manufacturer>]``
* ``AT#XCARRIERCFG="model_number"[,<model_number>]``
* ``AT#XCARRIERCFG="software_version"[,<version>]``
* ``AT#XCARRIERCFG="device_serial_no_type"[,<device_serial_no_type>]``
* ``AT#XCARRIERCFG="service_code"[,<service_code>]``
* ``AT#XCARRIERCFG="pdn_type"[,<pdn_type>]``
* ``AT#XCARRIERCFG="binding"[,<binding>]``
* ``AT#XCARRIERCFG="server_enable"[,<0|1>]``
* ``AT#XCARRIERCFG="is_bootstrap"[,<0|1>]``
* ``AT#XCARRIERCFG="lifetime"[,<lifetime>]``
* ``AT#XCARRIERCFG="sec_tag"[,<sec_tag>]``
* ``AT#XCARRIERCFG="uri"[,<uri>]``
* ``AT#XCARRIERCFG="session_idle_timeout"[,<session_idle_timeout>]``

The values of the parameters depend on the command string used.
If a valid command string is used without any parameter, the current value of the corresponding configuration will be returned in the response, as in the examples shown below.

> Boolean configurations, such as ``auto_startup`` or ``is_bootstrap``, only accept 0 (disable) and 1 (enable) as input parameters.

Reviewer: ```suggestion
Boolean configurations, such as ``auto_startup`` or ``is_bootstrap``, only accept ``0`` (disable) and ``1`` (enable) as input parameters.
```

The ``carriers`` configuration allows to choose the networks in which the LwM2M carrier library will apply.
Possible inputs are mapped as follows:

> * ``all`` - Any network allowed.

Reviewer: ```suggestion

* ``all`` - Any network allowed.
```
Empty line.

>   * ``all`` - Any network allowed.

Reviewer 2: Remove the indents here
